kuwala/core/canvas/src/data/state/models/CanvasModel.js

```diff
@@ -129,11 +152,15 @@ const CanvasModel = {
             targetDTO = targetElement.data.transformationBlock;
             targetDTO.connectedSourceNodeIds.push(source);
             updateTransformationBlock({ addNode, elements, setElements, updatedBlock: targetDTO });
+        } else if (targetElement.type === EXPORT_BLOCK) {
+            targetDTO = targetElement.data.exportBlock;
+            targetDTO.connectedSourceNodeIds.push(source);
+            updateExportBlock({ addNode, elements, setElements, updatedBlock: targetDTO });
```

Comment:
This action can be moved outside of all the blocks and only conditionally set the `targetDTO`.